sonaranalyzer-dotnet/its/expected/akka.net/Akka.MultiNodeTestRunner.Shared-{964F0EC5-FBE6-47C5-8AE6-145114D5DB8C}-S4056.json

```diff
@@ -134,175 +56,6 @@
 "id":  "S4056",
 "message":  "Use the overload that takes a 'CultureInfo' or 'IFormatProvider' parameter.",
 "location":  {
-"uri":  "sources\akka.net\src\core\Akka.MultiNodeTestRunner.Shared\Persistence\VisualizerRuntimeTemplate.Tree.cs",
-"region":  {
-"startLine":  111,
-"startColumn":  20,
-"endLine":  114,
-"endColumn":  39
-}
-}
-},
-{
-"id":  "S4056",
-"message":  "Use the overload that takes a 'CultureInfo' or 'IFormatProvider' parameter.",
-"location":  {
-"uri":  "sources\akka.net\src\core\Akka.MultiNodeTestRunner.Shared\Reporting\SpecRunCoordinator.cs",
-"region":  {
-"startLine":  26,
-"startColumn":  37,
-"endLine":  26,
-"endColumn":  84
-}
-}
-},
-{
-"id":  "S4056",
-"message":  "Use the overload that takes a 'CultureInfo' or 'IFormatProvider' parameter.",
-"location":  {
-"uri":  "sources\akka.net\src\core\Akka.MultiNodeTestRunner.Shared\Sinks\ConsoleMessageSinkActor.cs",
```

Comment:
many of the "lost" issues are `string.StartsWith` which is ok, but many are not raising on `string.Format("Results for {0}", data.FactName))` (like the current line on `ConsoleMessageSinkActor.cs`) - I added tests for `string.Format` and I don't understand the behavior

By looking inside the `mscorlib.dll` contents for `System.string` I see only

```

        public static String Format(IFormatProvider provider, String format, params object[] args);

        public static String Format(String format, params object[] args);

        public static String Format(String format, object arg0, object arg1, object arg2);

        public static String Format(String format, object arg0);

        public static String Format(String format, object arg0, object arg1);
```

Case in which it depends how Roslyn identifies the `invocation` - it may overlap with the `params` overload or not.

But the official docs from https://docs.microsoft.com/en-us/dotnet/api/system.string.format?view=netframework-4.7.2 show overloads for each of the above signatures, like below. If the `mscorlib.dll` would respect the official docs, we wouldn't have FN. And I don't want to add complexity to my implementation because of this.

```
Format(String, Object) 
Format(String, Object[])
Format(IFormatProvider, String, Object) 
Format(IFormatProvider, String, Object[])
Format(String, Object, Object)
Format(IFormatProvider, String, Object, Object)
Format(String, Object, Object, Object)
Format(IFormatProvider, String, Object, Object, Object)
```

Reply:
although `string.Format` is quite used and is quite the "predilect target" of this rule... so I'm not completely sure.
I would rather open a separate ticket for the FN rather than fixing it in this ticket...

Reply:
after discussing with @duncanp-sonar, I should fix the FNs before merging this ticket to avoid a bad experience for users